fix: set default value None for optional fields so we do not get validation error when the field is missing

umglurf opened the pull request:
## Description

After upgrade to pydantic2, optional fields needs to have default value None. Also adds fix for root_validator being replaced by model_validator

**Related issue (if applicable):** fixes #<issue number goes here>

## Checklist
  - [x] The code change is tested and works locally.
  - [ ] tests pass. **Your PR won't be merged unless tests pass**
  - [ ] There is no commented out code in this PR

umglurf commented:
Tests pass when https://github.com/flipper/pygrocy2/pull/3 is applied

cliffordwhansen commented:
@umglurf I did some testing (added all the things, chore, mealplan, shopping list, battery) and found the following are also needed:
```
diff --git a/pygrocy/grocy_api_client.py b/pygrocy/grocy_api_client.py
index 2bb7098..9828f71 100644
--- a/pygrocy/grocy_api_client.py
+++ b/pygrocy/grocy_api_client.py
@@ -36,7 +36,7 @@ class ShoppingListItem(BaseModel):
     id: int
     product_id: Optional[int] = None
     note: Optional[str] = None
-    amount: float
+    amount: Optional[float]
     row_created_timestamp: datetime
     shopping_list_id: int
     done: int
@@ -228,7 +228,7 @@ class CurrentBatteryResponse(BaseModel):
 class BatteryData(BaseModel):
     id: int
     name: str
-    description: str
+    description: Optional[str]
     used_in: str
     charge_interval_days: int
     created_timestamp: datetime = Field(alias="row_created_timestamp")
@@ -245,7 +245,7 @@ class BatteryDetailsResponse(BaseModel):
 
 class MealPlanSectionResponse(BaseModel):
     id: Optional[int] = None
-    name: str
+    name: Optional[str]
     sort_number: Optional[int] = None
     row_created_timestamp: datetime
```
Can you add these to your PR?

umglurf commented:
@cliffordwhansen pushed an update with these fixes now

marbletravis commented:
@umglurf it might be worth running this https://docs.pydantic.dev/2.10/migration/#install-pydantic-v2, I did so and had a few other changes. (you can see my PR, I think your is probably better at this point, didn't see these before I started working on this), however, I also had to change the datetime import, not sure if you ran into issues with that as well or if it was just me.

umglurf commented:
Hi @marbletravis . I ran into the datetime issue when trying to run the tests, but saw that it had been handled by #1 
I've added your other fixes into this PR and updated the title and description

marbletravis commented:
@umglurf nice work! Thanks.

dantol1 commented:
Any eta on when this will be merged and a new release will be created? 
And thanks for providing a solution this quick :)

cliffordwhansen commented:
Thanks for the update @flipper

I would be willing to help out, but can't take over, maybe a couple of us can be co-maintainers?

marbletravis commented:
I am also willing to help, I am a C# developer by day, learning python by night. @flipper would you be willing to accept this PR and push another version to PyPi, so we can get the fix out quicker?

flipper commented:
I realized i haven't pushed my changes. I already fixed this issue when i started my fork.
https://pypi.org/project/pygrocy2/2.3.3/

umglurf commented:
Hi @flipper 

These changes are not in 2.3.3. Is it easiest if you merge the 2.3.3 branch to develop and I rebase this PR so you can push out a 2.3.4 release with these fixes?

flipper commented:
https://pypi.org/project/pygrocy2/2.4.0/
